src/ledger/insert_shred.zig

```diff
@@ -176,6 +183,7 @@ pub const ShredInserter = struct {
         var newly_completed_data_sets = ArrayList(CompletedDataSetInfo).init(allocator);
         defer newly_completed_data_sets.deinit();
         for (shreds, is_repaired) |shred, is_repair| {
+            // QUESTION: What guarantees that contents of shred/is_repair are ordered same way?
```

Comment:
should be. In another way, it doesn't make sense

Reply:
It's expected that the caller structures them in the same order. If not, that's a bug, and you should not be surprised to see problems.

This is only used for metrics and logging. The only thing a bug here would do is mess with observability.